dnf/base.py

```diff
@@ -1516,7 +1516,7 @@ def package_downgrade(self, pkg):
         if not q:
             msg = _("Package %s not installed, cannot downgrade it.")
             logger.warning(msg, pkg.name)
-            return 0
+            raise dnf.exceptions.MarkingError(_('No match for argument: %s') % pkg.location, pkg.name)
```

Comment:
Didn't tested, but guessing that it will throw Unpack exception as on left side you have `%s`, but on right side you have more (but didn't check constructor of `dnf.exceptions.MarkingError`